File: Arithmetic_Triplets/arithmetic_triplets.py
from typing import List


class Solution:
    def arithmeticTriplets(self, nums: List[int], diff: int) -> int:
        if len(nums) < 3:
            return 0

        triplets = 0
        for num in nums[: len(nums) - 2]:
            equal_diff, compare_num = 0, num
            for next_num in nums[1:]:
                if next_num - compare_num == diff:
                    equal_diff += 1
                    compare_num = next_num

                if equal_diff == 2:
                    triplets += 1
                    break

        return triplets

File: Arithmetic_Triplets/arithmetic_triplets.go
package main

func arithmeticTriplets(nums []int, diff int) int {
	if len(nums) < 3 {
		return 0
	}

	triplets := 0
	for i := 0; i < len(nums)-2; i++ {
		diffEqual, compareNum := 0, nums[i]
		for j := i + 1; j < len(nums); j++ {
			if nums[j]-compareNum == diff {
				diffEqual++
				compareNum = nums[j]
			}

			if diffEqual == 2 {
				triplets++
				break
			}
		}
	}

	return triplets
}
